src/script/interpreter.cpp

```diff
@@ -1634,6 +1740,25 @@ static bool ExecuteWitnessScript(const Span<const valtype>& stack_span, const CS
 {
     std::vector<valtype> stack{stack_span.begin(), stack_span.end()};
 
+    // OP_SUCCESSx processing overrides everything, including stack element size limits
```

Comment:
this would be easy/nice to split out for unit tests

Reply:
This is a bit annoying, as this loop has multiple distinct return branches, and it returning success can mean either continue execution or not.

Happy to take some code if you have a suggestion, though.

Reply:
Yeah found out the hard way after trying a few times.